android/src/main/java/com/oblador/keychain/KeychainModule.java

```diff
@@ -316,6 +437,436 @@ protected void getGenericPassword(@NonNull final String alias,
+                        }
+
+                        @Override
+                        public void onAuthenticationSucceeded(FingerprintManager.AuthenticationResult result) {
+                          super.onAuthenticationSucceeded(result);
+                          if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                            putExtraWithAES(key, value, mSharedPreferences, false, strings, pm, result.getCryptoObject().getCipher());
+                          }
+                        }
+                      }, null);
+            }
+          }
+          return;
+        }
+
+        byte[] encryptedBytes = cipher.doFinal(value.getBytes());
+
+        // Encode the initialization vector (IV) and encryptedBytes to Base64.
+        String base64IV = Base64.encodeToString(cipher.getIV(), Base64.DEFAULT);
+        String base64Cipher = Base64.encodeToString(encryptedBytes, Base64.DEFAULT);
+
+        String result = base64IV + DELIMITER + base64Cipher;
+
+        putExtra(key, result, mSharedPreferences);
+        pm.resolve(value);
+      } catch (InvalidKeyException | UnrecoverableKeyException e) {
+        try {
+          mKeyStore.deleteEntry(KEY_ALIAS_AES);
+          prepareKey();
+        } catch (Exception keyResetError) {
+          pm.reject(keyResetError);
+        }
+        pm.reject(e);
+      } catch (IllegalBlockSizeException e){
+        if(e.getCause() != null && e.getCause().getMessage().contains("Key user not authenticated")) {
+          try {
+            mKeyStore.deleteEntry(KEY_ALIAS_AES);
+            prepareKey();
+            pm.reject(AppConstants.KM_ERROR_KEY_USER_NOT_AUTHENTICATED, e.getCause().getMessage());
+          } catch (Exception keyResetError) {
+            pm.reject(keyResetError);
+          }
+        } else {
+          pm.reject(e);
+        }
+      } catch (SecurityException e) {
+        pm.reject(e);
+      } catch (Exception e) {
+        pm.reject(e);
+      }
+    } else {
+      pm.reject(AppConstants.E_BIOMETRIC_NOT_SUPPORTED, "Biometrics not supported");
+    }
+  }
+
+
+  @ReactMethod
+  public void setItem(String key, String value, ReadableMap options, Promise pm) {
+
+    String name = sharedPreferences(options);
+
+    if (options.hasKey("touchID") && options.getBoolean("touchID")) {
+      boolean showModal = options.hasKey("showModal") && options.getBoolean("showModal");
+      HashMap strings = options.hasKey("strings") ? options.getMap("strings").toHashMap() : new HashMap();
+      putExtraWithAES(key, value, prefs(name), showModal, strings, pm, null);
+    } else {
+      try {
+        putExtra(key, encrypt(value), prefs(name));
+        pm.resolve(value);
+      } catch (Exception e) {
+        e.printStackTrace();
+        pm.reject(e);
+      }
+    }
+  }
+
+  @ReactMethod
+  public void deleteItem(String key, ReadableMap options, Promise pm) {
```

Comment:
Can we delete an item that has biometric-enabled?

Reply:
For deleting biometric-enabled value, we have resetGenericPassword API